cpp/models/lct_secir/model.h

```diff
@@ -35,233 +35,372 @@ namespace mio
+     * @return Result of the simulation divided in infection states without subcompartments. 
+     *  Returns TimeSeries with values -1 if calculation is not possible.
+     */
+    TimeSeries<ScalarType> calculate_compartments(const TimeSeries<ScalarType>& subcompartments_ts) const
+    {
+        Eigen::Index count_InfStates  = (Eigen::Index)InfectionState::Count;
+        Eigen::Index num_compartments = count_InfStates * m_groups;
+        TimeSeries<ScalarType> compartments_ts(num_compartments);
+        if (!(this->populations.get_num_compartments() == (size_t)subcompartments_ts.get_num_elements())) {
+            log_error("Result does not match infectionState of the Model.");
+            Eigen::VectorXd wrong_size = Eigen::VectorXd::Constant(num_compartments, -1);
+            compartments_ts.add_time_point(-1, wrong_size);
+            return compartments_ts;
+        }
+        Eigen::VectorXd compartments(num_compartments);
+        for (Eigen::Index timepoint = 0; timepoint < subcompartments_ts.get_num_time_points(); ++timepoint) {
+            compress_vector(subcompartments_ts[timepoint], compartments);
+            compartments_ts.add_time_point(subcompartments_ts.get_time(timepoint), compartments);
+        }
+
+        return compartments_ts;
+    }
+
+    /**
+     * @brief Checks that the model satisfies all constraints (e.g. parameter or population constraints).
+     * @return Returns true if one or more constraints are not satisfied, false otherwise.
+     */
+    bool check_constraints() const
+    {
 
-        auto params           = this->parameters;
-        auto total_population = pop.sum() - pop[LctState::template get_first_index<InfectionState::Dead>()];
+        return (check_constraints_impl() || this->parameters.check_constraints() ||
+                this->populations.check_constraints());
+    }
+
+private:
+    /**
+     * @brief Converts a vector with subcompartments in a vector without subcompartments 
+     * by summing up subcompartment values.
+     * This is done recusively for each group which corresponds to a slice of the vector.
+     *
+     * @tparam group The group specifying the slice of the vector being considered. 
+     * @param[in] subcompartments The vector that should be converted.
+     * @param[out] compartments Reference to the vector where the output is stored.
+     */
+    template <size_t Group = 0, std::enable_if_t<(Group < m_groups) && (Group >= 0), bool> = true>
+    void compress_vector(const Eigen::VectorXd& subcompartments, Eigen::VectorXd& compartments) const
+    {
+        using LctState = typename std::tuple_element_t<Group, tupleLctStates>;
 
-        ScalarType infectedNoSymptoms = 0;
-        ScalarType infectedSymptoms   = 0;
-        ScalarType flow               = 0;
+        // Specify first indices of the vector slices considered.
+        Eigen::Index first_index_subcomps = this->populations.template get_first_index_group<Group>();
```

Comment:
```suggestion
        Eigen::Index first_index_group = this->populations.template get_first_index_group<Group>();
```
Would be more intuitive for me.